core/Pos/Aeson/Core.hs

```diff
@@ -39,14 +49,56 @@ deriving instance ToJSON SlotCount
 instance FromJSON CoinPortion where
     parseJSON v = unsafeCoinPortionFromDouble <$> parseJSON v
 
+instance ToJSON CoinPortion where
+    toJSON = toJSON . coinPortionToDouble
+
 deriveFromJSON S.defaultOptions ''VssCertificate
 deriveFromJSON S.defaultOptions ''Millisecond
 deriveFromJSON S.defaultOptions ''Microsecond
 deriveFromJSON S.defaultOptions ''Second
 deriveFromJSON S.defaultOptions ''SoftforkRule
 deriveFromJSON S.defaultOptions ''BlockVersionData
+deriveToJSON   S.defaultOptions ''Microsecond
 
 deriving instance FromJSON Timestamp
+deriving instance ToJSON Timestamp
+
+-- Instances needed for wallet
+deriveFromJSON defaultOptions ''SlotId
+deriveFromJSON defaultOptions ''BlockCount
+deriveFromJSON defaultOptions ''LocalSlotIndex
+deriveFromJSON defaultOptions ''ChainDifficulty
+
+instance ToJSON Script where
+    toJSON Script{..} = object [
+        "scrVersion"    .= scrVersion,
+        "scrScript" .= JsonByteString scrScript ]
```

Comment:
I tend to agree with @gromakovsky 's comment below: why prefix the JSON fields with `scr` (or any other prefix)? . I think `version` and `script` looks way neater.